### DISCUSSION TITLE
Interested in working on the project through GSoC and more

### DISCUSSION THREAD

**Discussion by gucifer** [pytorch/ignite#1630] [General]
Hello there,
I am highly interested in working on the metrics module of this project and have started reading through the code base. I am a final year student in computer science. 
I have been doing research work in Image classification and data noise reduction for over a year. Currently, I am doing a thesis on Text Summarization and NLP. So I am fairly familiar with ROUGE, BLEU, METEOR and others.
I am also reading up on distributed implementation for the same. Any good resource other than the pytorch documentation that might help?
I am facing issue with string operations. Since pytorch does not have a tensor representation for list of strings or similar, the step of counting common strings(n-grams) does not seem to have a parallel alternative, unless each character is encoded using corresponding ASCII value as mentioned [here](https://discuss.pytorch.org/t/how-to-feed-string-data-to-a-gpu-to-encode-data-sentencetransformers/85572).

**Comment by vfdev-5** [comment]
Thanks for your interest in the project and GSoC application we submit ! 
Working on metics module would be really awesome ! Apart of GSoC way to collaborate we could also explore other ways (please, contact me on discord vfdev-5#4458 to discuss those modalities).


> I am facing issue with string operations. Since pytorch does not have a tensor representation for list of strings or similar, the step of counting common strings(n-grams) does not seem to have a parallel alternative, unless each character is encoded using corresponding ASCII value as mentioned here.

I think string encoding into numbers is a good approach. Currently, in ignite we provide a helper broadcast and all_gather methods to work on strings:
- https://pytorch.org/ignite/distributed.html#ignite.distributed.utils.broadcast
- https://pytorch.org/ignite/distributed.html#ignite.distributed.utils.all_gather

**Reply by gucifer** [reply]
Hi there, 
I send you a friend request on discord, my id is kana#7979. Let us discuss further on there!

**Reply by sdesrozis** [reply]
Hi @gucifer ! Thank you for your interest 😊 and good luck for your thesis, we definitely need guys on NLP !

**Comment by gucifer** [comment]
Hello there,
I have done a basic implementation of ROUGE Score, commit for the same can be seen on [my fork of ignite](https://github.com/gucifer/ignite/blob/master/ignite/metrics/rouge_n.py). It is a simple implementation and does not follow DDP, will add support for it once I learn how to.
Also, what is the mathematical notation being followed for the comments? Anything else I should be aware of before i open a PR?
